src/pages/GamePage.jsx

```diff
@@ -8,72 +8,57 @@ const ranArr = Array.from({ length: numbersQuantity }, () =>
   Math.floor(Math.random() * numbersRange)
 );
 const triesLimit = 5;
-const selectedArray = [];
-let filteredArray = [];
 
 const GamePage = () => {
   const [userNumber, setUserNumber] = useState(false),
     [errorHandler, setErrorHandler] = useState(false),
     [resultHandler, setResultHandler] = useState(false),
     [repeatError, setRepeatError] = useState(false),
     [triesCounter, setTriesCounter] = useState(0);
-
+  let resVal = false;
+  let checkVal = false;
+  let evenCheck = false;
   const inputHandler = (text) => {
     setUserNumber(parseInt(text.target.value));
   };
   const saveUserNumber = (event) => {
-    let checkVal = false;
+    if (ranArr[0] % 2 === 0) {
+      evenCheck = true;
+    }
     setTriesCounter(triesCounter + 1);
     if (isNaN(userNumber)) {
       setErrorHandler("Podaj cyfre");
     } else {
       event.preventDefault();
       setErrorHandler(false);
-      filteredArray = evenFilter(ranArr);
-      selectedArray.forEach((elem) => {
-        if (elem === userNumber) {
-          checkVal = true;
-        } else {
-          checkVal = false;
-        }
-      });
+      ranArr.forEach((elem, index, arr) =>
+        checkArray(elem, index, arr, userNumber, numbersQuantity)
+      );
       setRepeatError(checkVal);
       if (
         triesCounter < triesLimit &&
         resultHandler !== true &&
         checkVal !== true
       ) {
-        setResultHandler(iterArray(filteredArray, userNumber));
-        filteredArray.push(userNumber);
-        selectedArray.push(userNumber);
-        console.log(selectedArray);
+        setResultHandler(resVal);
+        ranArr.push(userNumber);
       }
     }
   };
-  const evenFilter = (array) => {
-    if (array[0] % 2 === 0) {
-      return array.filter((value) => {
-        return value % 2 === 0;
-      });
-    } else {
-      return array.filter((value) => {
-        return value % 2 !== 0;
-      });
-    }
-  };
-  const iterArray = (array, selectedNumber) => {
-    let i = 0;
-    while (array[i - 1] !== selectedNumber && i < array.length) {
-      i += 1;
-      console.log(i);
-    }
-    if (array[i - 1] === selectedNumber) {
-      return true;
-    } else {
-      return false;
+  const checkArray = (elem, index, arr, number, arrLength) => {
+    if (elem === number) {
+      if (index < arrLength) {
+        if (evenCheck === true && elem % 2 === 0) {
+          // wiem ze tutaj dubluje warunki ale uznalem ze tak bedzie prosciej i czytelniej
```

Comment:
sprawdzanie drugi raz jest niepotrzebne. Nie jest czytelniej, bo zastanawiasz się czemu to jest drugi raz